Sources/Filters/Sources/CubeSource/index.js

```diff
@@ -0,0 +1,243 @@
+
+      for (let j = 0; j < 2; j++) {
+        tc[0] = (x[0] + 0.5) * ((2 * i) - 1);
+        x[2] = model.center[2] - (model.zLength / 2.0);
+
+        for (let k = 0; k < 2; k++) {
+          tc[1] = (x[2] + 0.5) * (-1);
+
+          points[(pointIndex * 3)] = x[0];
+          points[(pointIndex * 3) + 1] = x[1];
+          points[(pointIndex * 3) + 2] = x[2];
+
+          normals[(pointIndex * 3)] = n[0];
+          normals[(pointIndex * 3) + 1] = n[1];
+          normals[(pointIndex * 3) + 2] = n[2];
+
+          textureCoords[(pointIndex * 2)] = tc[0];
+          textureCoords[(pointIndex * 2) + 1] = tc[1];
+
+          pointIndex++;
+          x[2] += model.zLength;
+        }
+        x[0] += model.xLength;
+      }
+      x[1] += model.yLength;
+      n[1] += 2.0;
+    }
+
+    x[2] = model.center[2] - (model.zLength / 2.0);
+    n[2] = (-1.0);
+    n[0] = n[1] = 0.0;
+    for (let i = 0; i < 2; i++) {
+      x[1] = model.center[1] - (model.yLength / 2.0);
+
+      for (let j = 0; j < 2; j++) {
+        tc[1] = x[1] + 0.5;
+        x[0] = model.center[0] - (model.xLength / 2.0);
+
+        for (let k = 0; k < 2; k++) {
+          tc[0] = (x[0] + 0.5) * ((2 * i) - 1);
+
+          points[(pointIndex * 3)] = x[0];
+          points[(pointIndex * 3) + 1] = x[1];
+          points[(pointIndex * 3) + 2] = x[2];
+
+          normals[(pointIndex * 3)] = n[0];
+          normals[(pointIndex * 3) + 1] = n[1];
+          normals[(pointIndex * 3) + 2] = n[2];
+
+          textureCoords[(pointIndex * 2)] = tc[0];
+          textureCoords[(pointIndex * 2) + 1] = tc[1];
+
+          pointIndex++;
+          x[0] += model.xLength;
+        }
+        x[1] += model.yLength;
+      }
+      x[2] += model.zLength;
+      n[2] += 2.0;
+    }
+
+    // Define quads
+    const polys = new window[model.pointType](numberOfPolys * 5);
+    polyData.getPolys().setData(polys, 1);
+
+    let polyIndex = 0;
+
+    polys[polyIndex++] = 4;
+    polys[polyIndex++] = 0;
+    polys[polyIndex++] = 1;
+    polys[polyIndex++] = 3;
+    polys[polyIndex++] = 2;
+
+    polys[polyIndex++] = 4;
+    polys[polyIndex++] = 4;
+    polys[polyIndex++] = 6;
+    polys[polyIndex++] = 7;
+    polys[polyIndex++] = 5;
+
+    polys[polyIndex++] = 4;
+    polys[polyIndex++] = 8;
+    polys[polyIndex++] = 10;
+    polys[polyIndex++] = 11;
+    polys[polyIndex++] = 9;
+
+    polys[polyIndex++] = 4;
+    polys[polyIndex++] = 12;
+    polys[polyIndex++] = 13;
+    polys[polyIndex++] = 15;
+    polys[polyIndex++] = 14;
+
+    polys[polyIndex++] = 4;
+    polys[polyIndex++] = 16;
+    polys[polyIndex++] = 18;
+    polys[polyIndex++] = 19;
+    polys[polyIndex++] = 17;
+
+    polys[polyIndex++] = 4;
+    polys[polyIndex++] = 20;
+    polys[polyIndex++] = 21;
+    polys[polyIndex++] = 23;
+    polys[polyIndex] = 22;
+  }
+
+  publicAPI.setBounds = (xMin, xMax, yMin, yMax, zMin, zMax) => {
+    const bounds = [xMin, xMax, yMin, yMax, zMin, zMax];
+    model.setBounds(bounds);
```

Comment:
You right, only one function can exist. But you can make it so both profiles can exist.

```
publicAPI.setBounds = (...bounds) => {
   let boundsArray = bounds;
   if (Arrays.isArray(bounds[0])) {
      boundsArray = bounds[0];
   }
   [...]
}
```